### PR TITLE
[bitnami/redis] Do not create master and replica serviceaccounts when using sentinel

### PR DESCRIPTION
**Pull request by wilmardo**
### Description of the change

Currently when choosing to use sentinel you get all the serviceaccounts not just the one needed.

Currently is creates the master and replica service accounts:
```
default          0         8h
redis1           0         73m
redis1-master    0         64s
redis1-replica   0         64s
redis2           0         51m
redis2-master    0         62s
redis2-replica   0         62s
```

### Benefits

Less clutter, removes unneeded resources. More inline with the other resources in the `master/` and `replica/` template folder. They all have this if check, this is the one resource where it was missing.

### Checklist

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [x] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami/readme-generator-for-helm)
- [x] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

**Comment by wilmardo** [comment]
I could add a check like this to the serviceaccount creation:
```
{{- if  and .Values.sentinel.enabled .Values.serviceAccount.create }}
```
Which would be more inline with the previous if statement but to me it makes more sense that the `ServiceAccount` creation only depends on the true/false of the `.Values.serviceAccount.create`. 
The `RoleBinding` references this `ServiceAccount` for example so then a working statement would be:

```
{{- if  and .Values.serviceAccount.create .Values.sentinel.enabled .Values.rbac.create }}
```
Again, just depending on the create true/false makes more sense to avoid edge cases like this in the future

**Comment by italoiz** [comment]
Is this PR a duplicate of https://github.com/bitnami/charts/pull/22223 ?

**Comment by Mauraza** [comment]
Hi @wilmardo,

Thanks for the contribution!! Could you check the conflicts?

**Comment by wilmardo** [comment]
@italoiz yes excuse me duplicated indeed. Dedupped the PR and changed the scope.

@Mauraza rebased and resolved the conflicts.

**Review by Mauraza — APPROVED** [review]
Hi @wilmardo,

Thanks for the contribution!!! LGTM!!

File: bitnami/redis/Chart.yaml [hunk 1]
```diff
@@ -34,4 +34,4 @@ maintainers:
 name: redis
 sources:
 - https://github.com/bitnami/charts/tree/main/bitnami/redis
-version: 18.8.1
+version: 18.8.2
```
Comment (Mauraza) [inline bitnami/redis/Chart.yaml:37]: Coul you update the version?
```suggestion
version: 18.8.3
```

**Pull request merged**